Stall after completing betterquests quest. [BUG]

**Describe the bug**
A clear and concise description of what the bug is:
After completing any betterquests quests if the quest reward is to run a command, the server will instantly freeze and stall the server.
We have tried with and without magma to which the problem on persists whilst magma is installed.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to betterquest quest book
2. Click on any quest that rewards with a command
3. Claim the reward
4. Enjoy the server stall

**Expected behavior**
A clear and concise description of what you expected to happen:
After completing the quest the server should run the command and the player should receive the reward.
**Info (please complete the following information):**
- Mod Name: BetterQuesting
 - Mod Version: BetterQuesting-3.5.329
 - Magma Version: Magma-c40b4af-server and Magma-d02c5b5-server
 - OS: Linux

Stall report link

https://hasteb.in/ozitivin.nginx

```cmd
[15:57:36] [Server thread/FATAL] [net.minecraft.server.MinecraftServer]: Error executing task
 java.util.concurrent.ExecutionException: java.lang.NullPointerException
        at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_262]
        at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_262]
        at net.minecraft.util.Util.func_181617_a(SourceFile:47) [h.class:?]
        at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:924) [MinecraftServer.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:456) [nz.class:?]
        at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:861) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:733) [MinecraftServer.class:?]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_262]
Caused by: java.lang.NullPointerException
        at org.bukkit.command.Command.testPermissionSilent(Command.java:207) ~[Command.class:966e691]
        at org.bukkit.command.Command.testPermission(Command.java:177) ~[Command.class:966e691]
        at net.minecraft.tileentity.CommandBlockBaseLogic.executeCommand(CommandBlockBaseLogic.java:281) ~[amj.class:?]
        at net.minecraft.tileentity.CommandBlockBaseLogic.executeSafely(CommandBlockBaseLogic.java:230) ~[amj.class:?]
        at net.minecraft.command.FunctionObject$CommandEntry.func_194145_a(FunctionObject.java:119) ~[bm$b.class:?]
        at net.minecraft.advancements.FunctionManager$QueuedCommand.func_194222_a(FunctionManager.java:218) ~[nt$a.class:?]
        at net.minecraft.advancements.FunctionManager.func_194019_a(FunctionManager.java:141) ~[nt.class:?]
        at bq_standard.rewards.RewardCommand.claimReward(RewardCommand.java:81) ~[RewardCommand.class:?]
        at betterquesting.questing.QuestInstance.claimReward(QuestInstance.java:223) ~[QuestInstance.class:?]
        at betterquesting.network.handlers.NetQuestAction.claimQuest(NetQuestAction.java:80) ~[NetQuestAction.class:?]
        at betterquesting.network.handlers.NetQuestAction.onServer(NetQuestAction.java:58) ~[NetQuestAction.class:?]
        at betterquesting.network.PacketQuesting$HandleServer.lambda$onMessage$0(PacketQuesting.java:76) ~[PacketQuesting$HandleServer.class:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_262]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_262]
        at net.minecraft.util.Util.func_181617_a(SourceFile:46) ~[h.class:?]
        ... 5 more
```

this is the log, currently testing this one out myself

--- a/patches/net/minecraft/command/FunctionObject.java.patch
+++ b/patches/net/minecraft/command/FunctionObject.java.patch
@@ -8,7 +8,7 @@
  import net.minecraft.util.ResourceLocation;
  
  public class FunctionObject
-@@ -108,7 +109,14 @@
+@@ -108,7 +109,19 @@
  
              public void execute(FunctionManager functionManagerIn, ICommandSender sender, ArrayDeque<FunctionManager.QueuedCommand> commandQueue, int maxCommandChainLength)
              {
@@ -20,6 +20,11 @@
 +                } else {
 +                    bukkitSender = CommandBlockBaseLogic.unwrapSender(sender);
 +                }
++
++                if (bukkitSender == null) {
++                    bukkitSender = sender.getCommandSenderEntity().getBukkitEntity();
++                }
++
 +                CommandBlockBaseLogic.executeSafely(sender, bukkitSender, this.command);
              }
  

